rasterio/windows.py

```diff
@@ -275,13 +341,18 @@ def evaluate(window, height, width):
     Window
         A new Window object with absolute index values.
     """
-    try:
-        r, c = window
-        assert len(r) == 2
-        assert len(c) == 2
-    except (ValueError, TypeError, AssertionError):
-        raise ValueError("invalid window structure; expecting ints"
-                         "((row_start, row_stop), (col_start, col_stop))")
+    if isinstance(window, tuple):
+        warn_window_deprecation()
+        try:
+            r, c = window
+            assert len(r) == 2
+            assert len(c) == 2
+        except (ValueError, TypeError, AssertionError):
+            raise ValueError("invalid window structure; expecting ints"
+                             "((row_start, row_stop), (col_start, col_stop))")
+    else:
+        r, c = window.toranges()
```

Comment:
It will likely make it easier to follow if these are fully unpacked to their component values, instead of then indexing them out below (e.g., `r[0]`).